arcgis-ios-sdk-samples/Geometry/List transformations/ListTransformationsViewController.swift

```diff
@@ -0,0 +1,190 @@
+    }
+    
+    // add a graphic with the given geometry, color and style to the graphics overlay
+    func addGraphic(_ geometry: AGSGeometry, color: UIColor, style: AGSSimpleMarkerSymbolStyle) {
+        let sms = AGSSimpleMarkerSymbol(style: style, color: color, size: 15.0)
+        graphicsOverlay.graphics.add(AGSGraphic(geometry: geometry, symbol: sms, attributes: nil))
+    }
+    
+    // set up our datumTransformations array
+    func setupTransformsList() {
+        guard let map = mapView.map,
+            let inputSR = originalGeometry.spatialReference,
+            let outputSR = map.spatialReference else { return }
+        
+        // if orderByMapExtent is on, use the map extent when retrieving the transformations
+        if orderByMapExtent.isOn {
+            datatumTransformations = AGSTransformationCatalog.transformationsBySuitability(withInputSpatialReference: inputSR, outputSpatialReference: outputSR, areaOfInterest: mapView.visibleArea?.extent)
+        }
+        else {
+            datatumTransformations = AGSTransformationCatalog.transformationsBySuitability(withInputSpatialReference: inputSR, outputSpatialReference: outputSR)
+        }
+        
+        defaultTransformation = AGSTransformationCatalog.transformation(forInputSpatialReference: inputSR, outputSpatialReference: outputSR)
+
+        // unselect selected row
+        if let selectedIndexPath = tableView.indexPathForSelectedRow {
+            tableView.deselectRow(at: selectedIndexPath, animated: true)
+        }
+        
+        // remove projected graphic from overlay
+        if let graphic = projectedGraphic() {
+            // we have the projected graphic, remove it (it's always the last one)
+            graphicsOverlay.graphics.remove(graphic)
+        }
+        
+        tableView.reloadData()
+    }
+    
+    func setPEDataURL() {
+        if let projectionEngineDataURL = FileManager.default
+            .urls(for: .documentDirectory, in: .userDomainMask)
+            .first?.appendingPathComponent("PEDataRuntime") {
+            do {
+                guard try projectionEngineDataURL.checkResourceIsReachable() else { return }
+                
+                // Normally, this method would be called immediately upon application startup before any other API method calls.
+                // So usually it would be called from AppDelegate.application(_:didFinishLaunchingWithOptions:), but for the purposes
+                // of this sample, we're calling it here.
+                try AGSTransformationCatalog.setProjectionEngineDirectory(projectionEngineDataURL)
+            } catch {
+                print("Could not load projection engine data.  See the README file for instructions on adding PE data to your app.")
+            }
+        }
+        
+        setupTransformsList()
+    }
+
+    @IBAction func oderByMapExtentValueChanged(_ sender: Any) {
+        setupTransformsList()
+    }
+    
+    func projectedGraphic() -> AGSGraphic? {
```

Comment:
The way this method is named and used is more like a property than a method. I also think the implementation might be simpler without the `graphic` variable:

```swift
var projectedGraphic: AGSGraphic? {
    if graphicsOverlay.graphics.count > 1 {
        return graphicsOverlay.graphics.lastObject as? AGSGraphic
    } else {
        return nil
    }
}
```

Let me know what you think.

Reply:
Yea, that works for me.  Not too much of a difference in my mind, though, but I went with your solution.